enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,736 @@
+   equal to the original CPU requests, so that CRI-O can use the value
+   to configure the CPU shares for the container.
+
+We will not change pods in a way that changes their quality-of-service
+class. So, we would not strip CPU requests unless they also have
+memory requests, because if we mutate the pod so that it has no CPU or
+memory requests the quality-of-service class of the pod would be
+changed automatically. Any labeled pod that is already BestEffort
+would be annotated using `0` as the value so that CRI-O will have an
+indicator to configure the CPU shares as BestEffort.
+
+#### Kubelet Changes
+
+Kubelet will be changed to look for a configuration file,
+`/etc/kubernetes/management-pinning`, to enable the management
+workload partitioning feature. The file should contain the cpuset
+specifier for the CPUs making up the management CPU pool.
+
+Kubelet will be changed so that when the feature is enabled, [when a
+static pod definition is
+read](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/file_linux.go)
+it is mutated in a way similar to the API server admission hook will
+mutate regular pods.
+
+Kubelet will be changed so that when the feature is enabled, it
+advertises a new extended resource of
+`cpu.workload.openshift.io/management`, representing all of the CPU
+capacity of the host (not just the management CPU pool).
+
+#### Installer Changes
+
+The installer will be changed to accept a new `workload.partitions`
+parameter for each [machine
+pool](https://github.com/openshift/installer/blob/master/docs/user/customization.md#machine-pools)
+with a list of workload types and their settings. For now, the only
+setting will be the name and the CPU sets to treat as separate from
+the standard shared pool. The name `workload.partitions` may change,
+based on discussion on the installer code changes.
+
+```yaml
+controlPlane:
+  name: master
+  workload:
+    partitions:
+      - name: management
+        cpuIDs: 0-1
+compute:
+  - name: worker
+    workload:
+      partitions:
+        - name: management
+          cpuIDs: 0-1
+```
+
+The default is empty, and when the value is empty the management
+workload partitioning feature is disabled.
+
+For the first version, we will only support 1 workload type with the
+name "management". That restriction may be lifted in the future.
+
+The `cpuIDs` value is a CPU set specifier for the CPUs to add to the
+isolated set used for management components, using the same cpuset
+syntax used elsewhere in kubernetes. Asking for explicit IDs instead
+of simply a count gives the user control in situations where they need
+specific CPUs to be available because they're close to accelerators,
+NICs, or other special hardware.
+
+The installer will be changed to generate an extra machine config
+manifest to configure CRI-O so that containers from pods with the
+`cpu.workload.openshift.io/management` annotation are run on the
```

Comment:
I expected to see "partition" in this (if partition is the name we settle on).

Reply:
@smarterclayton so would that be `cpu.partition.workload.openshift.io/management` (assuming we stick with "partition")?